numba/parfors/parfor_lowering.py

```diff
@@ -937,6 +1005,51 @@ def _create_gufunc_for_parfor_body(
     gufunc_txt += "def " + gufunc_name + \
         "(sched, " + (", ".join(parfor_params)) + "):\n"
 
+    globls = {"np": np}
+
+    # First thing in the gufunc, we reconstruct tuples from their
+    # individual parts, e.g., orig_tup_name = (part1, part2,).
+    # The rest of the code of the function will use the original tuple name.
+    for tup_var, exp_names in tuple_var_to_expanded_names.items():
+        tup_type = typemap[tup_var]
+        gufunc_txt += "    " + param_dict[tup_var]
+        # Determine if the tuple is a named tuple.
+        if (isinstance(tup_type, types.NamedTuple) or
+            isinstance(tup_type, types.NamedUniTuple)):
+            named_tup = True
+        else:
+            named_tup = False
+
+        if named_tup:
+            # It is a named tuple so try to find the global that defines the
+            # named tuple.
+            found = False
+            func_def = guard(get_definition, lowerer.func_ir, tup_var)
+            if config.DEBUG_ARRAY_OPT:
+                print("func_def:", func_def, type(func_def))
+            if (func_def is not None and
+                isinstance(func_def, ir.Expr) and
+                func_def.op == 'call'):
+                func_def = guard(get_definition, lowerer.func_ir, func_def.func)
+                if config.DEBUG_ARRAY_OPT:
+                    print("func definition:", func_def, type(func_def))
+                if func_def is not None:
+                    if isinstance(func_def, ir.Global) or isinstance(func_def, ir.FreeVar):
+                        gval = func_def.value
+                        globls[func_def.name] = gval
+                        #globls[func_def.name] = eval(gval.__module__ + "." + gval.__name__)
+                        found = True
+            if found is False and config.DEBUG_ARRAY_OPT:
```

Comment:
If `found` is `False` is this not a problem in itself? What would happen in the following code if so!?

Reply:
Yes, it is most likely going to fail if found is False (note that I changed this code so that "found" is gone but the logic is essentially the same).  If you want, tell me what error to raise here and I will but that will only improve the error message.

Reply:
How about something along the lines of `numba.core.errors.CompilerError("Could not find definition of <whatever it is looking for>", loc)`?

Reply:
This actually manifests as e.g. `Untyped global name 'mytype': cannot determine Numba type of <class 'numba.core.ir.UndefinedType'>` which is probably sufficient.

Reply:
Done.